src/state.rs

```diff
@@ -1,21 +1,18 @@
-#[cfg(feature = "email")]
 use crate::mailgun::MailgunClient;
 
 use axum::extract::FromRef;
 
 #[derive(FromRef, Clone)]
 pub struct State {
-    #[cfg(feature = "email")]
-    pub mailgun_client: MailgunClient,
+    pub mailgun_client: Option<MailgunClient>,
     pub for_garde: (),
 }
 
 pub async fn new() -> State {
     dotenv::dotenv().ok();
 
     State {
-        #[cfg(feature = "email")]
-        mailgun_client: MailgunClient::from(crate::CONFIG.mailgun.clone()),
+        mailgun_client: crate::CONFIG.mailgun.clone().try_into().ok(),
         for_garde: (),
     }
 }
```

Comment:
```suggestion
pub async fn new() -> State {
    dotenv::dotenv().ok();

    let mailgun_client = match crate::CONFIG.mailgun.clone().try_into() {
        Ok(client) => Some(client),
        Err(e) => {
            if crate::CONFIG.mailgun.disable {
                None
            } else {
                panic!("failed to initialize mailgun client: {}", e)
            }
        }
    };

    State {
        mailgun_client,
        for_garde: (),
    }
}
```
This is to handle the case where the `try_into` fails with any of the other errors than the `mailgun is disabled`.